packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -322,10 +322,16 @@ class NavigatorState extends State<Navigator> {
     super.initState();
     assert(config.observer == null || config.observer.navigator == null);
     config.observer?._navigator = this;
-    _push(config.onGenerateRoute(new RouteSettings(
-      name: config.initialRoute ?? Navigator.defaultRouteName,
-      isInitialRoute: true
-    )));
+
+    bool isFirst = true;
+    List<String> routes = _splitRoute(config.initialRoute ?? Navigator.defaultRouteName).toList();
+    for (String route in routes) {
+      _push(config.onGenerateRoute(new RouteSettings(
+        name: route,
+        isInitialRoute: isFirst
+      )));
+      isFirst = false;
```

Comment:
What happens if you pass true for `isInitialRoute` for all of these?  You don't want them to animate in either, which means you want them to all be "initial".  (We should probably rename isInitialRoute to something that makes sense in this context).